llvm/lib/CodeGen/SelectionDAG/FunctionLoweringInfo.cpp

```diff
@@ -64,11 +64,19 @@ static ISD::NodeType getPreferredExtendForValue(const Instruction *I) {
   // can be exposed.
   ISD::NodeType ExtendKind = ISD::ANY_EXTEND;
   unsigned NumOfSigned = 0, NumOfUnsigned = 0;
-  for (const User *U : I->users()) {
-    if (const auto *CI = dyn_cast<CmpInst>(U)) {
+  for (const Use &U : I->uses()) {
+    if (const auto *CI = dyn_cast<CmpInst>(U.getUser())) {
       NumOfSigned += CI->isSigned();
       NumOfUnsigned += CI->isUnsigned();
     }
+    if (const auto *CallI = dyn_cast<CallBase>(U.getUser())) {
+      const AttributeList &PAL = CallI->getAttributes();
+      AttributeSet Attrs = PAL.getParamAttrs(U.getOperandNo());
```

Comment:
Do we need to make sure the use is one of the call operands and not the callee of an indirect call? Not sure how we determine what Instructions `getPreferredExtendForValue` is called for.

Reply:
Yeah, I think need to make sure the use is one of the call operands.  So I want to add `if (!CallI->isArgOperand(&U)) continue;`  between line 72 and line 73.  Do you think that's okay?

Reply:
I added `if (!CallI->isArgOperand(&U)) continue;` in new commit. Please help to review. Thanks.